### PR TITLE
[LINALG] Add decomposition of `aten.dropout` op

### PR DESCRIPTION
**Pull request by Shukla-Gaurav**
- This commit adds decomposition of `aten.dropout` op. It also covers the
  training mode of the same op.
- It also adds lowering of `aten.sub.float` op.

Signed-Off-by: Gaurav Shukla <gaurav@nod-labs.com>

### CODE REVIEW

**Comment by pashu123** [comment]
@Shukla-Gaurav @cathyzhyi I think we have the implementation of aten.bernoulli so we can decompose dropout rather than identity-implementation.

**Comment by Shukla-Gaurav** [comment]
> @Shukla-Gaurav @cathyzhyi I think we have the implementation of aten.bernoulli so we can decompose dropout rather than identity-implementation.

Sure, will look into it. Thanks :)

**Comment by Shukla-Gaurav** [comment]
Now this also covers the training mode. Please review again @cathyzhyi @pashu123

**Review by cathyzhyi — APPROVED** [review]
looks good but let's wait for #654 to be merged first. Thanks!

File: e2e_testing/torchscript/basic.py [hunk 1]
```diff
@@ -950,11 +990,11 @@ def __init__(self):
     ])
 
     def forward(self, x):
-        return torch.dropout(x, 0.0, False)
+        return torch.dropout(x, 0.3, train=True) * 0
```
Comment (cathyzhyi) [inline e2e_testing/torchscript/basic.py:993]: Can the result be tested with mean, var?

**Comment by cathyzhyi** [comment]
Please go ahead to merge when you are ready.

**Comment by Shukla-Gaurav** [comment]
@cathyzhyi It looks like there is some issue with the build system? The latest push did not trigger the build.

**Comment by cathyzhyi** [comment]
I think the build works now. Could you rebase and try again?

**Pull request merged**